YoCode/Output/Output.cs

```diff
@@ -8,11 +8,22 @@ namespace YoCode
     class Output
     {
         IPrint outputWriter;
+        IErrorReporter errOutput;
         FeatureData featData;
 
-        public Output(IPrint printTo)
+        public Output(IPrint printTo, IErrorReporter errorReporter = null)
         {
             outputWriter = printTo;
+
+            if (errorReporter == null)
+            {
+                errOutput = new NullErrorObject();
+            }
+            else
+            {
+                errOutput = errorReporter;
+            }
+
```

Comment:
This block can be simplified to `errOutput = errorReporter ?? new NullErrorObject();`